dask_image/ndinterp/__init__.py

```diff
@@ -0,0 +1,171 @@
+        except for `output_chunks`.
+
+    Parameters
+    ----------
+    input : array_like (Numpy Array, Dask Array, ...)
+        The input array.
+    matrix : array (ndim,) or (ndim, ndim)
+        Transformation matrix.
+    offset : array (ndim,)
+        Transformation offset.
+    output_shape : array (ndim,), optional
+        The size of the array to be returned.
+    output_chunks : array (ndim,), optional
+        The chunks of the output Dask Array.
+
+    Returns
+    -------
+    affine_transform : Dask Array
+        A dask array representing the transformed output
+
+    """
+
+    if output_shape is None:
+        output_shape = input.shape
+
+    # process kwargs
+
+    # set default order to 1, warn if different
+    if 'order' in kwargs:
+        order = kwargs['order']
+        del kwargs['order']
+        if order > 1:
+            UserWarning('Currently, for order > 1 the output of'
+                        '`dask_image.ndinterp.affine_transform` can differ'
+                        'from that of `ndimage.affine_transform`')
+    else:
+        order = 1
+
+    # prefilter is not yet supported
+    if 'prefilter' in kwargs:
+        if kwargs['prefilter'] and order > 1:
+            UserWarning('Currently, `dask_image.ndinterp.affine_transform`'
+                        'does not support `prefilter=True`')
+        del kwargs['prefilter']
+
+    transformed = da.zeros(output_shape,
+                           dtype=input.dtype,
+                           chunks=output_chunks)
+
+    transformed = transformed.map_blocks(resample_chunk,
+                                         dtype=input.dtype,
+                                         input=input,
+                                         matrix=matrix,
+                                         offset=offset,
+                                         order=order,
+                                         func_kwargs=kwargs,
+                                         )
+
+    return transformed
+
+
+def resample_chunk(chunk, input, matrix, offset, order, func_kwargs, block_info=None):
+    """Resample a given chunk, used by `affine_transform`."""
+
+    n = chunk.ndim
+    input_shape = input.shape
+    chunk_shape = chunk.shape
+
+    chunk_offset = [i[0] for i in block_info[0]['array-location']]
+
+    chunk_edges = np.array([i for i in np.ndindex(tuple([2] * n))])\
+        * np.array(chunk_shape) + np.array(chunk_offset)
+
+    rel_input_edges = np.dot(matrix, chunk_edges.T).T + offset
+    rel_input_i = np.min(rel_input_edges, 0)
+    rel_input_f = np.max(rel_input_edges, 0)
+
+    # expand relevant input to contain footprint of the spline kernel
+    # according to
+    #
+    # https://github.com/scipy/scipy/blob/
+    # 9c0d08d7d11fc33311a96d2ac3ad73c8f6e3df00/
+    # scipy/ndimage/src/ni_interpolation.c#L412-L419
+    #
+    # Also see https://github.com/dask/dask-image/
+    # issues/24#issuecomment-706165593
+    for dim in range(n):
+        if order % 1:
```

Comment:
Can you leave a short (in-line?) comment for future programmers describing why we have this modulo checking here.

Reply:
Good that you mentioned this. I took over the code @grlee77 posted here https://github.com/dask/dask-image/issues/24#issuecomment-706165593, but when comparing it to the [c code](https://github.com/scipy/scipy/blob/9c0d08d7d11fc33311a96d2ac3ad73c8f6e3df00/scipy/ndimage/src/ni_interpolation.c#L412-L419%20#%20noqa:%20E501) I think there might be some problems with it. I changed it to follow the c code now, however as commented I'm still unsure about a `-1` which is required to pass tests (that I extended to contain more edge cases).

Reply:
It does seem that which cases fail if the `- 1` is removed is not consistent from run to run of the tests (because the tests do not fix a random seed, different random offsets are tested each time).

However, I have only seen cases where `interp_order = 0` fail. All other orders always seem to pass without the ` - 1`.